### PR TITLE
Fix `dab validate` env_var parsing.

### PR DESCRIPTION
**Pull request by seantleonard**
## Why make this change?

- Closes #2200, an issue with `dab validate -c "path_to_config.json"` where environment variable `@env('ENV_VAR_NAME')`  references are unresolved during config schema validation and schema validation always fails. The issue specifically calls out an error when an environment variable is used for the property `database-type` can't resolve an environment variable and properly set an enum value (mssql, pgsql, mysql, etc) for the property.

## What is this change?

- This change corrects how `RuntimeConfigValidator::ValidateConfigSchema(...)` resolves environment variables in the runtime config. The usage of `string jsonData = runtimeConfig.ToJson();` converts the runtime config object (which already has environment variables resolved) into valid json which can then be validated against `dab.draft.schema.json`.

## How was this tested?

- [x] Integration Tests: added a test which exercises the code path that validates the dab runtime config against the dab json schema. By ensuring that no schema validation errors occur due to unresolved environment variables. 

## Sample Request(s)

Add the file `launchSettings.json` to `src/Cli/Properties` with the following contents:
```json
{
  "profiles": {
    "start": {
      "commandName": "Project",
      "commandLineArgs": "start -c src\\Cli\\dab-config.json",
      "environmentVariables": {
        "DATABASE_TYPE": "mssql",
        "DATABASE_NAME": "<your DB name",
        "CONN_STRING": "<CONN_STRING>",
        "AUTH_METHOD": "Simulator",
        "PUB_ENT_SRC": "publishers"
      },
      "httpPort": 5002
    },
    "validate": {
      "commandName": "Project",
      "commandLineArgs": "validate -c src\\Cli\\dab-config.json",
      "environmentVariables": {
        "DATABASE_TYPE": "mssql",
        "DATABASE_NAME": "<your DB name>",
        "CONN_STRING": "<CONN_STRING>",
        "AUTH_METHOD": "Simulator",
        "PUB_ENT_SRC": "publishers"
      },
      "httpPort": 5002
    }
  }
}
```

and modify your dab-config.json to have the following contents (assumes that you've deployed the DAB test project's .sql script:

```json
{
  "$schema": "https://github.com/Azure/data-api-builder/releases/download/v0.13.0-rc/dab.draft.schema.json",
  "data-source": {
    "database-type": "@env('DATABASE_TYPE')",
    "connection-string": "@env('CONN_STRING')"
  },
  "runtime": {
    "rest": {
      "enabled": true,
      "path": "/api"
    },
    "graphql": {
      "allow-introspection": true,
      "enabled": true,
      "path": "/graphql"
    },
    "host": {
      "mode": "development",
      "cors": {
        "origins": [
          "http://localhost:5000"
        ],
        "allow-credentials": false
      },
      "authentication": {
        "provider": "@env('AUTH_METHOD')"
      }
    }
  },
  "entities": {
    "Query": {
      "source": "books",
      "permissions": [
        {
          "role": "anonymous",
          "actions": [
            "read"
          ]
        }
      ],
      "graphql": {
        "type": {
          "singular": "ProductCatalogue",
          "plural": "ProductCatalogue"
        }
      }
    },
    "Product": {
      "source": "@env('PUB_ENT_SRC')",
      "permissions": [
        {
          "role": "authenticated",
          "actions": [
            "read",
            "delete",
            "update"
          ]
        }
      ]
    }
  }
}
```

### CODE REVIEW

File: src/Cli.Tests/ValidateConfigTests.cs [hunk 1]
```diff
@@ -107,4 +113,55 @@ public void TestValidateWithEmptyConfig()
             Assert.Fail($"Unexpected Exception thrown: {ex.Message}");
         }
     }
+
+    /// <summary>
+    /// This method implicitly validates that RuntimeConfigValidator::ValidateConfigSchema(...) successfully
+    /// executes against a config file referencing environment variables.
+    /// [CLI] ConfigGenerator::IsConfigValid(...)
+    ///     |_ [Engine] RuntimeConfigValidator::TryValidateConfig(...)
+    ///        |_ [Engine] RuntimeConfigValidator::ValidateConfigSchema(...)
```
Comment (Aniruddh25) [inline src/Cli.Tests/ValidateConfigTests.cs:122]: Where is the assert that verifies it doesn't execute successfully?

**Review by Aniruddh25 — APPROVED** [review]
LGTM, thanks for the quick fix!

**Review by aaronburtle — APPROVED** [review]
Looks good to me!

File: src/Cli.Tests/ValidateConfigTests.cs (hunk 1 above)
Comment (seantleonard) [reply src/Cli.Tests/ValidateConfigTests.cs:122]: added additional assert that ensures config schema validation failure messages aren't emitted.
```csharp
        Assert.IsFalse(
            condition: loggerOutput.Contains("Failed to validate config against schema due to"),
            message: "Unexpected errors encountered when validating config schema in RuntimeConfigValidator::ValidateConfigSchema(...).");
```

**Pull request merged**